src/libYARP_dev/include/yarp/dev/IFrameSource.h

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (C) 2006-2018 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-3-Clause license. See the accompanying LICENSE file for details.
+ */
+
+#ifndef YARP_DEV_IFRAMESOURCE_H
+#define YARP_DEV_IFRAMESOURCE_H
+
+#include <unordered_set>
+#include <map>
+#include <vector>
+#include <functional>
+#include <robotology/Result.h>
+#include <yarp/dev/api.h>
+#include <yarp/math/Quaternion.h>
+#include <yarp/math/FrameTransform.h>
+
+namespace yarp {
+    namespace dev {
+        class IFrameSource;
+    }
+}
+
+struct CustomFrameCompare
+{
+    bool operator()(const yarp::math::FrameTransform& lhs, const yarp::math::FrameTransform& rhs)
+    {
+        if (lhs.frameId == rhs.frameId)
+            return true;
+    }
+};
+
+class FrameEditor
+{
+    std::map<std::string, yarp::math::FrameTransform> storage;
+    yarp::math::FrameTransform nullframe;
+public:
+    const auto begin() const { return storage.begin(); }
+    const auto end() const { return storage.end(); }
+    bool insertUpdate(const yarp::math::FrameTransform& frame)
+    {
+        storage[frame.frameId] = frame;
+        return true;
+    }
+
+    robotology::Result<yarp::math::FrameTransform> get(const std::string& frameid)
+    {
+        if (storage.find(frameid) == storage.end())
+        {
+            return { false, 0, "", nullframe };
+        }
+        return { true, 0, "", storage[frameid] };
+    }
+};
+
+class YARP_dev_API yarp::dev::IFrameSource
+{
+    enum ConnectionType
+    {
+        DIRECT,
+        INVERSE,
+        UNDIRECT,
+        DISCONNECTED
+    };
+    ConnectionType getConnectionType(const std::string& target_frame, const std::string& source_frame, std::string* commonAncestor = nullptr);
+
+    robotology::Result<yarp::math::FrameTransform> getChainedTransform(const std::string& target_frame_id, const std::string& source_frame_id);
+
+    virtual bool frameExistsRaw(const std::string &frame_id);
```

Comment:
Why `raw` in the name?

Reply:
the "raw" version of the method are internal only and are not guarded. BTW see the "naming" note in the notes paragraph of the pr

Reply:
Not sure which "naming" note you are referring to. If it is "there will be, for sure, someone who don't like the names of the new flags/interfaces/something and has better alternative. not a problem for me. Just decide the names you like and i will change it." it is not that I do not like it, I just did not understand why `Raw` was added, as typically it is used in "***Raw" interfaces.